### PR TITLE
Create method for TopicsController

### PR DESCRIPTION
**Pull request by AnyaMaz**
#367 
- [x] Create Method
- [x] Tests sor create method
- [x] Mapping  Topic -> TopicModel

### CODE REVIEW

**Comment by AnyaMaz** [comment]
@malinink Rdy For Review

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 1]
```diff
@@ -62,6 +67,21 @@
...
      */
+
+    @PostMapping("")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Identity create(@RequestBody Topic topic) throws BadRequestException {
+        if (topic.getId() != null) {
+            throw new IdentifiedModelException();
+        }
+
+        final TopicModelInterface topicModel = this.topicsMapper.map(topic, TopicModel.class);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: checkSubject -> VR

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java [hunk 2]
```diff
@@ -133,6 +141,52 @@ public void testListForbidden() throws Exception {
...
+                ))
+                .andExpect(status().is(201))
+                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(content().json(
+                        new JsonSupport()
+                                .with(id, 5)
+                                .build()
+                ))
+                .andReturn();
+        verify(this.topicsService, times(1)).save(argumentCaptor.capture());
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java:270]: use support class

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java [hunk 3]
```diff
@@ -133,6 +141,52 @@ public void testListForbidden() throws Exception {
...
+                        .with(id, 2)
+                        .with(name, "TopicName")
+                        .with(subject, new JsonSupport().with(id, 1))
+                        .build()
+                ))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().string(""))
+                .andReturn();
+    }
+
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java:288]: add checkSubjects clause case test

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java (hunk 2 above)
Comment (AnyaMaz) [reply src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java:270]: done

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java (hunk 3 above)
Comment (AnyaMaz) [reply src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java:288]: done

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 4]
```diff
@@ -62,6 +66,34 @@
...
+        return this.topicsMapper.map(topicModel, Identity.class);
+    }
+
+    private void checkSubject(Topic topic) throws ForbiddenException {
+        final SubjectModelInterface subjectModel = this.subjectsService.find(
+                topic.getSubject().getId(),
+                this.subjectsOptionsBuilder.forAuth()
+                                );
+        if (!this.acl.hasAccess(subjectModel)) {
+            throw new ForbiddenException();
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: bad request

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java [hunk 5]
```diff
@@ -133,6 +140,75 @@ public void testListForbidden() throws Exception {
                 .andReturn();
 
     }
+//    TopicModelEmpty
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java]: remove comment

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java [hunk 6]
```diff
@@ -133,6 +140,75 @@ public void testListForbidden() throws Exception {
...
+                .andExpect(status().is(201))
+                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(content().json(
+                        new JsonSupport()
+                                .with(id, 5)
+                                .build()
+                ))
+                .andReturn();
+        verify(this.topicsService, times(1)).save(argumentCaptor.capture());
+        this.topicsSupport.assertEquals(argumentCaptor.getValue(), topicAdditionalEntity);
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java]: swap values, topicAdditionalEntity is an expected value

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by AnyaMaz** [comment]
@malinink Ready for review

**Review by malinink — APPROVED** [review]

**Pull request merged**